src/plone/restapi/tests/test_services_types.py

```diff
@@ -105,3 +107,58 @@ def test_file_type(self):
         self.assertIn('fieldsets', response)
         self.assertIn(
             'file.data', response['properties']['file']['properties'])
+
+    def test_addable_types_for_non_manager_user(self):
+        user = api.user.create(
+            email='noam.chomsky@example.com',
+            username='noam',
+            password='1234'
+        )
+
+        folder = api.content.create(
+            container=self.portal,
+            id="folder",
+            type='Folder',
+            title=u'folder',)
+
+        folder_cant_add = api.content.create(
+            container=self.portal,
+            id="folder_cant_add",
+            type='Folder',
+            title=u'folder_cant_add',)
+
+        api.user.grant_roles(
+            user=user,
+            obj=folder,
+            roles=['Contributor', ])
+
+        api.user.grant_roles(
+            user=user,
+            obj=folder_cant_add,
+            roles=['Reader', ])
+
+        transaction.commit()
+
+        self.api_session.auth = ('noam', '1234')
+        # In the folder, the user should be able to add types since we granted
+        # Contributor role on it
+        response = self.api_session.get('/folder/@types')
+        response = response.json()
+
+        self.assertIn(
+            'Document', [a['title'] for a in response if a['addable']])
+
+        # In the folder where the user only have Reader role, no types are
```

Comment:
@sneridagh I prefer to not have assert statements in a test and then do another action. When this happens I try to split up the tests, so that each tests only has assert statements at the end (precondition, action, assert). This is just a matter of preferences though.